rulewerk-vlog/src/main/java/org/semanticweb/rulewerk/reasoner/vlog/VLogInMemoryDataSource.java

```diff
@@ -0,0 +1,107 @@
+package org.semanticweb.rulewerk.reasoner.vlog;
+
+/*-
+ * #%L
+ * Rulewerk VLog Reasoner Support
+ * %%
+ * Copyright (C) 2018 - 2020 Rulewerk Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.Arrays;
+
+import org.semanticweb.rulewerk.core.reasoner.implementation.InMemoryDataSource;
+import org.semanticweb.rulewerk.core.reasoner.implementation.DataSourceConfigurationVisitor;
+
+/**
+ * Implementation of {@link InMemoryDataSource} for the VLog backend.
+ */
+public class VLogInMemoryDataSource extends InMemoryDataSource {
+	String[][] data;
+	int nextEmptyTuple = 0;
+
+	public VLogInMemoryDataSource(final int arity, final int initialCapacity) {
+		super(arity, initialCapacity);
+		this.data = new String[initialCapacity][arity];
+	}
+
+	/**
+	 * Transforms a constant name in a format suitable for the
+	 * reasoner. The default implementation assumes the VLog backend.
+	 * @param constantName the name of the constant
+	 *
+	 * @return a transformed version of constantName that is suitable for the Reasoner.
+	 */
+	protected String transformConstantName(String constantName) {
```

Comment:
To avoid duplicated code, a method from TermToVLogConverter can be used. Note that three methods that transform a constant name to a VLog-specific String representation exist:
- TermToVLogConverter#getVLogNameForConstant
- TermToVLogConverter#getVLogNameForConstantName - that is only used in unit tests; I think this can be used by VLogInMemoryDataSource
- VLogInMemoryDataSource#transformConstantName

We should avoid duplicated code if possible